sklearn/tests/test_common.py

```diff
@@ -225,51 +226,52 @@ def fit(self, X, y=None, **params):
         return super().fit(X, y, **params)
 
 
-def test_strict_mode_check_estimator():
-    # Tests various conditions for the strict mode of check_estimator()
+def test_api_only_check_estimator():
+    # Tests various conditions for the api_only parameter of check_estimator()
     # Details are in the comments
 
-    # LogisticRegression has no _xfail_checks, so when strict_mode is on, there
+    # LogisticRegression has no _xfail_checks, so when api_only=False, there
     # should be no skipped tests.
     with pytest.warns(None) as catched_warnings:
-        check_estimator(LogisticRegression(), strict_mode=True)
+        check_estimator(LogisticRegression(), api_only=False)
     assert not any(isinstance(w, SkipTestWarning) for w in catched_warnings)
-    # When strict mode is off, check_n_features should be skipped because it's
-    # a fully strict check
-    msg_check_n_features_in = 'check_n_features_in is fully strict '
-    with pytest.warns(SkipTestWarning, match=msg_check_n_features_in):
-        check_estimator(LogisticRegression(), strict_mode=False)
+    # When api_only is True, check_fit2d_1sample should be skipped
+    # because it's not an API check
+    skip_match = 'check_fit2d_1sample is not an API check'
+    with pytest.warns(SkipTestWarning, match=skip_match):
+        check_estimator(LogisticRegression(), api_only=True)
 
     # NuSVC has some _xfail_checks. They should be skipped regardless of
-    # strict_mode
+    # api_only
     with pytest.warns(SkipTestWarning,
                       match='fails for the decision_function method'):
-        check_estimator(NuSVC(), strict_mode=True)
-    # When strict mode is off, check_n_features_in is skipped along with the
-    # rest of the xfail_checks
-    with pytest.warns(SkipTestWarning, match=msg_check_n_features_in):
-        check_estimator(NuSVC(), strict_mode=False)
-
-    # MyNMF will fail check_fit_non_negative() in strict mode because it yields
-    # a bad error message
+        check_estimator(NuSVC(), api_only=False)
+    # When api_only is True, check_fit2d_1sample is skipped along
+    # with the rest of the xfail_checks
+    with pytest.warns(SkipTestWarning, match=skip_match):
+        check_estimator(NuSVC(), api_only=True)
```

Comment:
What I mean here is, as a user, if I call `check_estimator(..., api_only=True)`, I'm explicitly saying I only want the `api_only` tests. Then why would I get a warning which says `check_fit2d_1sample is not an API check`? Should I just not be getting that warning?

Reply:
We're treating these checks as if they were in the `_xfail_checks` tag  and for this tag we decided to ignore with a warning. It makes things much simpler to re-use that logic rather than implement a new check-ignoring logic (I believe there are discussions in previous PRs like #17361)

I personally don't have a strong opinion on whether the ignored checks should raise a warning or not. Note that for `parametrize_with_checks`, which is sort of the sister of `check_estimators`, those checks would be marked as XFAIL, so pytest would still output them in stdout saying something like "this was xfailed" or "this was xpassed".

Maybe @rth @thomasjpfan can remind us why we decided to ignore with a warning instead of just ignoring?

Reply:
I think this is better to raise a skip warning. It allows us to know which tests were not run and for which reason.